templates/glance/bin/init.sh

```diff
@@ -17,35 +17,61 @@ set -ex
 
 # This script generates the glance-api.conf/logging.conf file and
 # copies the result to the ephemeral /var/lib/config-data/merged volume.
-#
-# Secrets are obtained from ENV variables.
-export DBPASSWORD=${DatabasePassword:?"Please specify a DatabasePassword variable."}
-# TODO
-#export TRANSPORTURL=${TransportUrl:?"Please specify a TransportUrl variable."}
-export GLANCEPASSWORD=${GlancePassword:?"Please specify a GlanceKeystoneAuthPassword variable."}
+
+export PASSWORD=${GlancePassword:?"Please specify a GlanceKeystoneAuthPassword variable."}
 export DBHOST=${DatabaseHost:?"Please specify a DatabaseHost variable."}
 export DBUSER=${DatabaseUser:-"glance"}
 export DB=${DatabaseName:-"glance"}
+export DBPASSWORD=${DatabasePassword:?"Please specify a DatabasePassword variable."}
+
 
+DEFAULT_DIR=/var/lib/config-data/default
+CUSTOM_DIR=/var/lib/config-data/custom
+MERGED_DIR=/var/lib/config-data/merged
 SVC_CFG=/etc/glance/glance-api.conf
-SVC_CFG_MERGED=/var/lib/config-data/merged/glance-api.conf
-
-# expect that the common.sh is in the same dir as the calling script
-SCRIPTPATH="$( cd "$(dirname "$0")" >/dev/null 2>&1 ; pwd -P )"
-. ${SCRIPTPATH}/common.sh --source-only
-
-# Copy default service config from container image as base
-cp -a ${SVC_CFG} ${SVC_CFG_MERGED}
-
-# Merge all templates from config-data
-for dir in /var/lib/config-data/default; do
-    merge_config_dir ${dir}
-done
-
-# set secrets
-# TODO: transportUrl (either set here or elsewhere)
-#crudini --set ${SVC_CFG_MERGED} DEFAULT transport_url $TRANSPORTURL
-crudini --set ${SVC_CFG_MERGED} database connection mysql+pymysql://${DBUSER}:${DBPASSWORD}@${DBHOST}/${DB}
-crudini --set ${SVC_CFG_MERGED} keystone_authtoken password $GLANCEPASSWORD
-# TODO: transportUrl (either set here or elsewhere)
-#crudini --set ${SVC_CFG_MERGED} oslo_messaging_notifications transport_url $TRANSPORTURL
+SVC_CFG_MERGED=${MERGED_DIR}/glance-api.conf
+SVC_CFG_MERGED_DIR=${MERGED_DIR}/glance.conf.d
+
+mkdir -p ${SVC_CFG_MERGED_DIR}
+
+cp ${DEFAULT_DIR}/* ${MERGED_DIR}
+
+# Save the default service config from container image as glance-api.conf.sample,
+# and create a small glance-api.conf file that directs people to files in
+# glance.conf.d.
+
+cp -a ${SVC_CFG} ${SVC_CFG_MERGED}.sample
+cat <<EOF > ${SVC_CFG_MERGED}
+# Service configuration snippets are stored in the glance.conf.d subdirectory.
+EOF
+
+cp ${DEFAULT_DIR}/glance-api.conf ${SVC_CFG_MERGED_DIR}/00-default.conf
+
+# Generate 01-deployment-secrets.conf
+DEPLOYMENT_SECRETS=${SVC_CFG_MERGED_DIR}/01-deployment-secrets.conf
+
+cat <<EOF >> ${DEPLOYMENT_SECRETS}
+[database]
+connection = mysql+pymysql://${DBUSER}:${DBPASSWORD}@${DBHOST}/${DB}
```

Comment:
Yes, the long term plan is to remove entirely the `init.sh` script and generate the config within the operator: this will result in simplifying the operator as we don't have to run the initContainer anymore.
However, this requires some experiments, so let's say this is the first step to reach that goal.